hermes-console-vue/src/store/favorites/useFavorites.ts

```diff
@@ -0,0 +1,43 @@
+import { defineStore } from 'pinia';
+import { subscriptionFqn } from '@/utils/subscription-utils/subscription-utils';
+import type { FavoritesState } from '@/store/favorites/types';
+
+export const useFavorites = defineStore('favorites', {
+  state: (): FavoritesState => {
+    return {
+      topics: [],
+      subscriptions: [],
+    };
+  },
+  actions: {
+    async addTopic(topic: string) {
+      this.topics.push(topic);
+    },
+    async addSubscription(topic: string, subscription: string) {
+      this.subscriptions.push(subscriptionFqn(topic, subscription));
+    },
+    async removeTopic(topic: string) {
+      this.topics = this.topics.filter((topicName) => topicName !== topic);
+    },
+    async removeSubscription(topic: string, subscription: string) {
+      const subscriptionQualifiedName = subscriptionFqn(topic, subscription);
+      this.subscriptions = this.subscriptions.filter(
+        (subscriptionName) => subscriptionName !== subscriptionQualifiedName,
+      );
+    },
+    async removeSubscriptionByQualifiedName(subscriptionQualifiedName: string) {
+      this.subscriptions = this.subscriptions.filter(
+        (subscriptionName) => subscriptionName !== subscriptionQualifiedName,
+      );
+    },
+  },
+  getters: {
+    getTopics(state: FavoritesState): string[] {
+      return state.topics?.sort((a, b) => a.localeCompare(b));
```

Comment:
suggestion: if the order in which favourites were added is not important (they are sorted lexicographically here), we can use `Set` instead of array. This would also prevent duplicated record if there were some weird edge cases